_playwright-tests/UI/RBACUserTest.spec.ts

```diff
@@ -0,0 +1,76 @@
+// _playwright-tests/UI/RBAC.spec.ts
+import { test, expect } from '@playwright/test';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { randomName, randomUrl } from './helpers/repoHelpers';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+import { getUserAuthToken } from '../helpers/loginHelpers';
+
+const repoNamePrefix = 'Repo-RBAC';
+const repoName = `${repoNamePrefix}-${randomName()}`;
+const url = randomUrl();
+
+// Define a test group for admin user
+test.describe('Admin User Tests', () => {
+  test.use({ storageState: '.auth/admin_user.json' });
+
+  test('Login as user 1 (admin) and manage repo', async ({ page }) => {
+    await test.step('Create a repository', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+
+      await page.getByLabel('Name').fill(repoName);
+      await page.getByLabel('Introspect only').click();
+      await page.getByLabel('URL').fill(url);
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+    });
+
+    await test.step('Read the repo', async () => {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible();
+      await row.getByLabel('Kebab toggle').click();
+      await page.getByRole('menuitem', { name: 'Edit' }).click();
+      await expect(page.getByRole('dialog', { name: 'Edit custom repository' })).toBeVisible();
+      await expect(page.getByPlaceholder('Enter name', { exact: true })).toHaveValue(repoName);
+      await expect(page.getByPlaceholder('https://', { exact: true })).toHaveValue(url);
+    });
+
+    await test.step('Update the repository', async () => {
+      await page.getByPlaceholder('Enter name', { exact: true }).fill(`${repoName}-Edited`);
+      await page.getByRole('button', { name: 'Save changes', exact: true }).click();
+    });
+  });
+
+  // Define a separate test group for read-only user
+  test.describe('Read-Only User Tests', () => {
+    test.use({
+      storageState: '.auth/read-only.json',
+    });
+    test.describe.configure({ mode: 'serial' });
```

Comment:
good idea, makes sense to not run the read only test if the admin one fails, since this one depends on the admin one 👍🏼 
but this is in the wrong describe block, in needs to be one level up in the "admin user tests" describe (line 14), that describe could also probably benefit from a better name 😅